cmake/BundledOSSPUUID.cmake

```diff
@@ -21,8 +21,8 @@ function(use_bundled_osspuuid SOURCE_DIR BINARY_DIR)
     message("Using bundled ossp-uuid")
 
     # Define patch step
-    set(PC "${Patch_EXECUTABLE}" -p1 -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-mac-fix.patch" &&
-           "${Patch_EXECUTABLE}" -p1 -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-no-prog.patch")
+    set(PC "${Patch_EXECUTABLE}" -p1 -N -r- -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-mac-fix.patch" &&
+           "${Patch_EXECUTABLE}" -p1 -N -r- -i "${SOURCE_DIR}/thirdparty/ossp-uuid/ossp-uuid-no-prog.patch" || :)
```

Comment:
That `|| :` is not very nice -- is it there because of the "patch already applied" error we sometimes get?  I haven't seen that error too many times, and I'd rather do a clean rebuild (until we can fix the problem properly) than possibly ignore real errors thrown by `patch`.

Reply:
That was me, I admit it's not a nice solution. The intention was to stop patch from throwing errors when it has already been applied, but the only way was to stop it from throwing errors altogether.

I got an idea that maybe instead of `|| :` at the end, we could try reverse applying with `--dry-run` to not actually change anything. This would verify that either we could apply the patch or it could be reverse applied, i.e. it has already been applied. Would this be a good enough workaround for you or do you want to revert the whole change?

Reply:
Note for @arpadboda: A clean build works on gentoo without the changes in this file.

Reply:
I'm OK with a change that ignores "already applied" errors only and not all errors.  I would also be OK with reverting this part of the change, so we just live with the "already applied" errors until we figure out how to fix them properly.